app/controllers/users_controller.rb

```diff
@@ -1,10 +1,25 @@
 class UsersController < ActionController::Base
   def index
     @users = User.all
-    # @current_user = current_user
   end
 
   def show
-    @user = User.find(params[:id])
+    if (params[:id] = 'sign out')
+      sign_out_and_redirect
+    else
+      find_user
+    end
+  end
+
+  private
+
+  def sign_out_and_redirect
+    sign_out current_user
+    redirect_to new_user_session_path
+  end
+
+  def find_user
+    @user = User.find_by(id: params[:id])
+    redirect_to user_path, alert: 'User is not available' unless @user
   end
```

Comment:
- [ ] Your devise is working really well :+1: but I guess while implementing the devise in the users_controller show method :fearful: Because when I try to go to this page then it takes me back to sign in page :confused: I am really confused with this behavior of your app :thinking: Kindly fix this issue to make your app work as expected :blush: